### ISSUE TITLE
Would a rewrite be something nice ?

### ISSUE THREAD

**Issue by loprima-l** [NixOS/nixos-search#850]
Hi everyone!

I was wondering if a frontend rewrite with a more modern tool like vue would be likeable.
I assume that elm was chosen for performance, but I don't think that frontend performance will be that bad on vue.
My take is that while on the backend looking for a 4ms performance tweak is useful. Frontend performance can be worse if it helps to get more feature done faster and a better UI.

I've had a look at #412, it's an effortless PR on Vue but looks complicated to me on ELM. But should significantly improve the user experience. And that's just one of many such issues.

So to summarise a bit I have 3 questions: Is it mandatory to use Elm? Would you like a rewrite in a language with a better DX and a lower learning curve? And if so, why?

Thank you very much !

**Comment by dasJ** [comment]
cc @garbas

**Comment by loprima-l** [comment]
@garbas Excuse me for the ping but I would love to hear your toughs on it !

**Comment by mweinelt** [comment]
Garbas left the marketing team and has become unavailable.

**Comment by Jaculabilis** [comment]
I do backend, not frontend, but I made an experimental foray into Vue to see how hard it would be to get something usable. [In the `vue` branch under `frontend-ng`](https://github.com/Jaculabilis/nixos-search/tree/vue), I have the basic layout of the search reproduced. It still needs work to get search filters and query parameters working, and the styling could use a pass from someone with more area knowledge.

![Image](https://github.com/user-attachments/assets/d7a6d2a4-c9c9-4c2f-97b2-8aeb4a5eb4c6)

@raboof What do you think about the idea?

**Comment by raboof** [comment]
I personally prefer Elm to Vue tbh - not because of performance but because the compiler's guardrails are helpful.

**Comment by Jaculabilis** [comment]
I have it configured with TypeScript, so it does have some compiler-time checks. I don't know how they compare to Elm's, since I haven't been able to get VS Code's Elm extension to provide any intellisense yet.

I don't know how widespread Elm knowledge is compared to Vue or other frameworks, but it may be worth considering if only for the sake of making it easier to accept contributions.
